subprojects/process-services/src/main/java/org/gradle/process/ProcessForkOptions.java

```diff
@@ -27,30 +28,31 @@
      *
      * @return The executable.
      */
+    @Nullable
     String getExecutable();
 
     /**
      * Sets the name of the executable to use.
      *
-     * @param executable The executable. Must not be null.
+     * @param executable The executable.
      * @since 4.0
      */
-    void setExecutable(String executable);
+    void setExecutable(@Nullable String executable);
 
     /**
      * Sets the name of the executable to use.
      *
-     * @param executable The executable. Must not be null.
+     * @param executable The executable.
      */
-    void setExecutable(Object executable);
+    void setExecutable(@Nullable Object executable);
 
     /**
      * Sets the name of the executable to use.
      *
-     * @param executable The executable. Must not be null.
```

Comment:
Why do we allow a `null` `executable` now?

Reply:
@wolfs Because the getter is `@Nullable` so the setter must also be otherwise the types won't line up correctly, see https://github.com/gradle/gradle/blob/807ee86c2a841209748b03cfec3514a25146fafb/subprojects/core/src/main/java/org/gradle/api/tasks/AbstractExecTask.java#L169-L173

Reply:
My point is that the getter should not be `@Nullable`. Where do we need `null` as an executable?

Reply:
It is effectively nullable in all implementations.
We don't "need null as an executable", but if you get it without setting it first it'll effectively be null.

Reply:
I stay with my opinion that `executable` cannot be `null`: https://github.com/gradle/gradle/blob/030c638f73f49b78163515fa2a925176341cbbdf/subprojects/core/src/main/java/org/gradle/process/internal/AbstractExecHandleBuilder.java#L129-L132

It should not be `Optional`. Actually, it should be a `Provider`, so you can ask if it has been set at some point and then base your logic on that. Or even better, use `map`.

I guess without using `Provider` we need to use `@Nullable`? I don't understand the `@Optional` though...